openmmtools/cache.py

```diff
@@ -387,8 +388,10 @@ def get_context(self, thermodynamic_state, integrator=None):
                 # Only one match.
                 context = self._lru[matching_context_ids[0]]
             else:
-                # Multiple matches, prefer non-default Integrator.
-                for context_id in matching_context_ids:
+                # Multiple matches, prefer the non-default Integrator.
+                # Always pick the least recently used to make two consective
+                # calls retrieving the same integrator.
+                for context_id in reversed(matching_context_ids):
                     if context_id[1] != self._default_integrator_id():
                         context = self._lru[context_id]
                         break
```

Comment:
Okay, so let me run through the list of conditions to make sure its all handled I don't miss anything:

1. No context: pick the default (main `if`)
2. One context (could be the default), pick the one (`elif`)
3. Multiple contexts, may be default, try to pick the non-default (`else`)

Is there ever going to be a case where multiple contexts are picked and they are all the default integrator? I know that each context/integrator pair is unique, but the default can just be cloned and fed into multiple contexts. I don't know if by this point the default is paired up and can't be used again.